src/main/java/seedu/task/logic/commands/SelectCommand.java

```diff
@@ -1,44 +1,74 @@
 
-    public static final String COMMAND_WORD = "select";
+	private boolean isTask;
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Selects the person identified by the index number used in the last person listing.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_SELECT_TASK_SUCCESS = "Selected Task: %1$s";
+	public static final String COMMAND_WORD = "select";
 
-    public SelectCommand(int targetIndex) {
-        this.targetIndex = targetIndex;
-    }
 
-    @Override
-    public CommandResult execute() {
+	public static final String MESSAGE_USAGE = COMMAND_WORD + "\n"
+			+ "Selects an existing task/event from the TaskBook.\n\n"
+			+ "Selects a task at the specified INDEX in the most recent task listing.\n"
+			+ "Parameters: SELECT_TYPE + INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + " -t"
+			+ " 1\n\n" + "Selects a event at the specified INDEX in the most recent event listing.\n"
+			+ "Parameters: SELECT_TYPE + INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + " -e"
+			+ " 1";
 
-        UnmodifiableObservableList<ReadOnlyTask> lastShownList = model.getFilteredTaskList();
+	public static final String MESSAGE_SELECT_TASK_SUCCESS = "Selected Task: %1$s";
+	public static final String MESSAGE_SELECT_EVENT_SUCCESS = "Selected Event: %1$s";
 
-        if (lastShownList.size() < targetIndex) {
-            indicateAttemptToExecuteIncorrectCommand();
-            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
-        }
+	
 
-        EventsCenter.getInstance().post(new JumpToListRequestEvent(targetIndex - 1));
-        return new CommandResult(String.format(MESSAGE_SELECT_TASK_SUCCESS, targetIndex));
+	public SelectCommand(Integer targetIndex, boolean isTask) {
+		// TODO Auto-generated constructor stub
+		this.targetIndex = targetIndex;
+		this.isTask = isTask;
+	}
 
-    }
+	@Override
+	public CommandResult execute() {
 
+		UnmodifiableObservableList<ReadOnlyTask> lastShownTaskList = model.getFilteredTaskList();
+		UnmodifiableObservableList<ReadOnlyEvent> lastShownEventList = model.getFilteredEventList();
+
+		if (isTask == true) {
+			if (lastShownTaskList.size() < targetIndex) {
+				indicateAttemptToExecuteIncorrectCommand();
+				return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+			}
+			
+			else {
+				EventsCenter.getInstance().post(new JumpToListRequestEvent(targetIndex - 1));
+				return new CommandResult(String.format(MESSAGE_SELECT_TASK_SUCCESS, targetIndex));
+			}
+		}
+		
+		if (isTask == false) {
+			if (lastShownEventList.size() < targetIndex) {
+				indicateAttemptToExecuteIncorrectCommand();
+				return new CommandResult(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+			}
+		}
+			
+			EventsCenter.getInstance().post(new JumpToListRequestEvent(targetIndex - 1));
```

Comment:
I guess the general logic is correct here. 
Do keep in mind that the select for task is actually implemented after we morphed the task card from person card. 

It is the event panel handling which u might need to add on, starting from posting a different but similar event. 

I suggest to take a look at task.ui.UIManager, as well as TaskListPanel and TaskListCard. 

Let's discuss about this feature tomorrow.